Wangscape/noise/Reseedable.h

```diff
@@ -43,6 +44,46 @@ struct Reseedable final
     Reseedable scalePoint(double x_scale, double y_scale, double z_scale);
     Reseedable translatePoint(double x_displace, double y_displace, double z_displace);
     Reseedable turbulence(double frequency, double power, int roughness, int seed);
+    template<typename Iterator>
+    Reseedable terrace(Iterator begin, Iterator end, bool inverted = false);
+    Reseedable terrace(int controlPointCount, bool inverted = false);
+    template<typename Iterator>
+    Reseedable curve(Iterator begin, Iterator end);
+private:
+    Reseedable finaliseTerrace(std::shared_ptr<module::Terrace>& terrace_ptr);
+    Reseedable finaliseCurve(std::shared_ptr<module::Curve>& curve_ptr);
 };
 
+template<typename Iterator>
+inline Reseedable Reseedable::terrace(Iterator it, Iterator end, bool inverted)
```

Comment:
Yes, my initial thought was to pass the container by const reference. But user code may want to use only part of a container, or an iterator that doesn't point to a container. I think the `std` method is best.

I'll fix the disagreement in parameter names between declaration and implementation, though.